lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts

```diff
@@ -3,17 +3,127 @@
+import { CryptoProvider } from "../../crypto/CryptoProvider";
+import {
+    ManagedIdentityErrorCodes,
+    createManagedIdentityError,
+} from "../../error/ManagedIdentityError";
+import {
+    HttpMethod,
+    METADATA_HEADER_NAME,
+    ManagedIdentityIdType,
+} from "../../utils/Constants";
 
+const ARC_API_VERSION: string = "2019-11-01";
+
+/**
+ * Original source of code: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/AzureArcManagedIdentitySource.cs
+ */
 export class AzureArc extends BaseManagedIdentitySource {
+    private endpoint: string;
+
+    constructor(
+        logger: Logger,
+        cacheManager: CacheManager,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider,
+        endpoint: string
+    ) {
+        super(logger, cacheManager, networkClient, cryptoProvider);
+
+        this.endpoint = endpoint;
+    }
+
+    public static tryCreate(
+        logger: Logger,
+        cacheManager: CacheManager,
+        networkClient: INetworkModule,
+        cryptoProvider: CryptoProvider
+    ): AzureArc | null {
+        const imdsEndpoint: string | undefined = process.env["IMDS_ENDPOINT"];
+
+        const [areEnvironmentVariablesValidated, endpoint]: [
+            boolean,
+            string | undefined
+        ] = validateEnvironmentVariables(
+            process.env["IDENTITY_ENDPOINT"] || undefined,
+            imdsEndpoint,
+            logger
+        );
+
+        return areEnvironmentVariablesValidated
+            ? new AzureArc(
+                  logger,
+                  cacheManager,
+                  networkClient,
+                  cryptoProvider,
+                  endpoint as string
+              )
+            : null;
+    }
+
     public createRequest(
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        _resource: string,
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        _managedIdentityId: ManagedIdentityId
+        resource: string,
+        managedIdentityId: ManagedIdentityId
     ): ManagedIdentityRequestParameters {
-        throw new Error("Method not implemented.");
+        if (
+            managedIdentityId.idType !== ManagedIdentityIdType.SYSTEM_ASSIGNED
+        ) {
+            throw createManagedIdentityError(
+                ManagedIdentityErrorCodes.unableToCreateAzureArc
+            );
+        }
+
+        const request: ManagedIdentityRequestParameters =
+            new ManagedIdentityRequestParameters(HttpMethod.GET, this.endpoint);
+
+        request.headers[METADATA_HEADER_NAME] = "true";
+        request.queryParameters["api-version"] = ARC_API_VERSION;
+        request.queryParameters["resource"] = resource;
+        // bodyParameters calculated in BaseManagedIdentity.acquireTokenWithManagedIdentity
+
+        return request;
     }
 }
+
+const validateEnvironmentVariables = (
+    identityEndpoint: string | undefined,
```

Comment:
A string can be `null`. Isn't it a better solution than using `undefined` which seems to be more lax?

Reply:
this is just verifying that the environment variable exists. It either exists, or it doesn't and therefore it's undefined. If the environment exists as null string, then that's bad, and this code will correctly register that as it not existing.